src/uu/uniq/src/uniq.rs

```diff
@@ -247,6 +250,41 @@ fn opt_parsed<T: FromStr>(opt_name: &str, matches: &ArgMatches) -> UResult<Optio
     })
 }
 
+/// Gets number of fields to be skipped from the shorthand option `-N`
+///
+/// ```bash
+/// uniq -12345
+/// ```
+/// the first digit isn't interpreted by clap as part of the value
+/// so `get_one()` would return `2345`, then to get the actual value
+/// we loop over every possible first digit, only one of which can be
+/// found in the command line because they conflict with each other,
+/// append the value to it and parse the resulting string as usize,
+/// an error at this point means that a character that isn't a digit was given
+fn obsolete_skip_field(matches: &ArgMatches) -> UResult<Option<usize>> {
+    for opt_text in OBSOLETE_SKIP_FIELDS_DIGITS {
+        let argument = matches.get_one::<String>(opt_text);
+        if matches.contains_id(opt_text) {
+            let mut full = opt_text.to_owned();
+            if let Some(ar) = argument {
+                full.push_str(ar);
+            }
+            let value = full.parse::<usize>();
+
+            if let Ok(val) = value {
+                return Ok(Some(val));
+            } else {
+                return Err(USimpleError {
```

Comment:
Could you add a test for this case too? Something like `uniq -500invalid`

Reply:
just added it, thanks for the advice